platform-tests/src/test/java/org/junit/platform/commons/util/ClasspathScannerTests.java

```diff
@@ -181,21 +181,28 @@ void scanForClassesInPackage() {
 	@Test
 	// #2500
 	void scanForClassesInPackageWithinModuleSharingNames() throws Exception {
-		var jarfile = getClass().getResource("/com.greetings@1-ea.jar");
+		var greetings = getClass().getResource("/com.greetings@1-ea.jar").toURI();
+		var tests = getClass().getResource("/com.greetings.test@0-ea.jar").toURI();
 
-		var module = "com.greetings";
+		var root = "com.greetings.test";
 		var before = ModuleFinder.of();
-		var finder = ModuleFinder.of(Path.of(jarfile.toURI()));
+		var finder = ModuleFinder.of(Path.of(greetings), Path.of(tests));
 		var boot = ModuleLayer.boot();
-		var configuration = boot.configuration().resolveAndBind(before, finder, Set.of(module));
+		var configuration = boot.configuration().resolve(before, finder, Set.of(root));
 		var parent = ClassLoader.getPlatformClassLoader();
 		var layer = ModuleLayer.defineModulesWithOneLoader(configuration, List.of(boot), parent).layer();
 
-		var classpathScanner = new ClasspathScanner(() -> layer.findLoader(module), ReflectionUtils::tryToLoadClass);
-
-		var classes = classpathScanner.scanForClassesInPackage("com.greetings", allClasses);
-		var classNames = classes.stream().map(Class::getName).collect(Collectors.toList());
-		assertThat(classNames).hasSize(1).contains("com.greetings.Main");
+		var classpathScanner = new ClasspathScanner(() -> layer.findLoader(root), ReflectionUtils::tryToLoadClass);
+		{
+			var classes = classpathScanner.scanForClassesInPackage("com.greetings", allClasses);
+			var classNames = classes.stream().map(Class::getName).collect(Collectors.toList());
+			assertThat(classNames).hasSize(2).contains("com.greetings.Main", "com.greetings.test.Tests");
+		}
+		{
+			var classes = classpathScanner.scanForClassesInPackage("com.greetings.test", allClasses);
+			var classNames = classes.stream().map(Class::getName).collect(Collectors.toList());
```

Comment:
nit: I'd use a static import for `toList()`.

Reply:
Let's wait until we switched to Java 16 -- and replace all those with `Stream::toList`

https://twitter.com/sormuras/status/1337495352962904066

Reply:
All? All 44 usages found in test files.

![image](https://user-images.githubusercontent.com/2319838/105299227-d1fd1180-5bba-11eb-9100-69ef933090bd.png)

Reply:
Go for it! 👍